### ISSUE TITLE
First time granting access to microphone, breaks microphone for all other apps too (qt 5.12 issue?)

### ISSUE THREAD

**Issue by luksus42** [ubports/morph-browser#459]
Devices: Nexus 5, Fairphone 3, BQ e4.5
Channel: dev

I assume this happens since QT 5.12 update.
If your html5-app or Morph-Browser for example, already did have permission to use the microphone or are in the permissions-list displayed at least, then everything works as expected.

### Expected Behavior
- Opening html5 app / open any microphone recording page in Morph 
- website asks for permission
- when choosing "yes", system asks for permission if it is not already in the systems allowed apps.

### Current Behavior
- Starting html5 recording test app / opening any microphone recording page in Morph
- website asks for permission
  - when choosing "yes": 
    - system does not ask for permission, the app is not saved in the permissions-list for the microphone
    - No other App can use the microphone until a lomiri restart (recorder App for example)
  - when choosing "no":
    - the app is not saved in the permissions-list for the microphone
    - microphone is still usable by other apps

Here is a test html5 app:
[html5rectest.zip](https://github.com/ubports/morph-browser/files/5954267/html5rectest.zip)

related: https://github.com/luksus42/soundrecog/issues/6
maybe related: #455 , #399

**Comment by dobey** [comment]
I don't think this is related to Qt 5.12. Even on stable, Recorder app is not asking for mic permissions on my E4.5. The mic permissions request in Morph also never completed for me. In System Settings, microphone permissions is only showing Guitar Tools as having requested permission.

After reboot, I was able to grant mic permission to Recorder. I started recording, paused, and then switched to Morph and opened a video call page to test. It asked for permissions in the browser, but did not open the system permissions request, and failed to complete the request. I switched back to Recorder, and unpaused the recording, which resulted in some odd audio playback and recording feedback loop, which persisted until I switched back to Morph, where it again popped up the in-browser request and not the system request. So I think Morph is doing something wrong and perhaps causing the trust service to crash or such.

**Comment by luksus42** [comment]
Thanks for your quick answer @dobey .
So something other must have changed in the last two weeks...

**Comment by peat-psuwit** [comment]
Ok, so turns out the problem is a bit complicated.

It seems like the Chromium version used in the current QtWebEngine now issues a recording in a child process instead of the main process (perhaps this changes in QtWebEngine 5.12 -> 5.14.). This causes Mir, when being requested a prompt session, to unable to match the recording pid to the opened session.

This should results in the request being simply denied, but the reason it hangs is that it's the prompt itself that hangs. The trust store daemon ignores the error and proceeding with launching the prompt provider, which fails to connect to Mir and then hangs. This hang cascade all the way to Pulseaudio, which is the reason why future audio recording also fails.

As this isn't exactly Morph Browser's problem, I'll transfer this issue to ubports/ubuntu-touch.
